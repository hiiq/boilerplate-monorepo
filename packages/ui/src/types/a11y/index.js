import { Role } from './role';

export const A11y = {
  Role,
};
