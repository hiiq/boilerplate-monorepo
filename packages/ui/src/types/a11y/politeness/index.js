import { Enumeration, propTypes } from './typedef';

export const Politeness = {
  ...Enumeration,
  propTypes,
};
