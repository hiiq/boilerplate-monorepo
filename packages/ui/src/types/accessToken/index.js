// This variable is the only place that the access token is stored
let token = null;

const clear = () => {
  token = null;
};
const read = () => {
  return token;
};
const set = newToken => {
  token = newToken;
};

export const AccessToken = {
  clear,
  read,
  set,
};
