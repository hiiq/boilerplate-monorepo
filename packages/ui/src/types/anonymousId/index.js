import { LocalStorage } from '@boilerplate-monorepo/ui-common';
import cuid from 'cuid';

const makeId = () => `anonymous-${cuid()}`;
const clear = () => localStorage.removeItem(LocalStorage.ANONYMOUS_ID);

const set = () => {
  const newId = makeId();

  localStorage.setItem(LocalStorage.ANONYMOUS_ID, newId);

  return newId;
};

const read = () => localStorage.getItem(LocalStorage.ANONYMOUS_ID) || set();

export const AnonymousId = {
  clear,
  read,
  set,
};
