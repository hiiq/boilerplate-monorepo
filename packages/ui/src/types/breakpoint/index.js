export const Breakpoint = {
  HUGE: 1440,
  LARGE: 1170,
  MEDIUM: 768,
  SMALL: 450,
};
