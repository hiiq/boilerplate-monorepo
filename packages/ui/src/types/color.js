export const Color = {
  black: '#212121',
  border: '#DCDCDC',
  error: '#DB0000',
  focusRing: '#7AACFE',
  muted: '#8D8D8D',
  primary: '#6d4baa',
  success: '#0C9C0C',
  warn: '#DFDF00',
  white: '#FFFFFF',
};
