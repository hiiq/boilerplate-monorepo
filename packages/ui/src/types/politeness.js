import { oneOf } from 'prop-types';
import { values } from 'ramda';

export const Politeness = {
  ASSERTIVE: 'assertive',
  POLITE: 'polite',
};

export const propTypes = oneOf(values(Politeness));
