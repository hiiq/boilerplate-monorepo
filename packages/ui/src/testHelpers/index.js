import { render } from '@testing-library/react';
import { node } from 'prop-types';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components/macro';
import { Theme } from 'types/theme';

export * from '@testing-library/react';

jest.mock('shared/useTheme');
jest.mock('shared/useTranslate');

const AllTheProviders = ({ children }) => (
  <Router>
    <ThemeProvider theme={Theme.example()}>{children}</ThemeProvider>
  </Router>
);

AllTheProviders.propTypes = {
  children: node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export { customRender as render };
