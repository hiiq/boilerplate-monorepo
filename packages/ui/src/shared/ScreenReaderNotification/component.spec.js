import React from 'react';
import { render } from 'testHelpers';
import { Politeness } from 'types/politeness';
import { ScreenReaderNotification, domTestId } from '.';

describe('<ScreenReaderNotification/>', () => {
  it('renders properly', () => {
    const defaultProps = { children: <x-child /> };
    const { getByTestId } = render(
      <ScreenReaderNotification {...defaultProps} />
    );
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('renders politeness properly', () => {
    const defaultProps = {
      children: <x-child />,
      politeness: Politeness.ASSERTIVE,
    };
    const { getByTestId } = render(
      <ScreenReaderNotification {...defaultProps} />
    );
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });
});
