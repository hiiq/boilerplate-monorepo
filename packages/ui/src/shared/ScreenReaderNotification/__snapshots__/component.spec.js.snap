// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenReaderNotification/> renders politeness properly 1`] = `
<div
  aria-live="assertive"
  class="ScreenReaderNotification__Styled-o4wxu2-0 hZyAuJ"
  data-testid="ScreenReaderNotification"
>
  <x-child />
</div>
`;

exports[`<ScreenReaderNotification/> renders properly 1`] = `
<div
  aria-live="polite"
  class="ScreenReaderNotification__Styled-o4wxu2-0 hZyAuJ"
  data-testid="ScreenReaderNotification"
>
  <x-child />
</div>
`;
